src/CustomerProvider/DefaultCustomerProvider.php

```diff
@@ -208,6 +208,7 @@ public function getActiveCustomerByEmail($email)
             return null;
         }
 
+        $email = str_replace('_', '\_', $email);
```

Comment:
please change line: `$list->addConditionParam('trim(email) like ?', [$list->escapeLike(trim($email))]);` and revert this. thanks!